npm/pkg/dataplane/dataplane_windows_test.go

```diff
@@ -0,0 +1,629 @@
+		Name:       setMetadata.GetPrefixName(),
+		PolicyType: pType,
+		Values:     strings.Join(members, ","),
+	}
+}
+
+// verifyHNSCache asserts that HNS has the correct state.
+// TODO: move all these functions to common location used by windows test files in pkg ipsets and policies.
+func verifyHNSCache(t *testing.T, hns *hnswrapper.Hnsv2wrapperFake, expectedSetPolicies []*hcn.SetPolicySetting, expectedEndpointACLs map[string][]*hnswrapper.FakeEndpointPolicy) {
+	// we want to evaluate both verify functions even if one fails, so don't write as verifySetPolicies() && verifyACLs() in case of short-circuiting
+	success := verifySetPolicies(t, hns, expectedSetPolicies)
+	success = verifyACLs(t, hns, expectedEndpointACLs) && success
+
+	printGetAllOutput(hns)
+	if !success {
+		require.FailNow(t, fmt.Sprintf("hns cache had unexpected state. printing hns cache...\n%s", hns.Cache.PrettyString()))
+	}
+}
+
+// verifySetPolicies is true if HNS strictly has the expected SetPolicies.
+func verifySetPolicies(t *testing.T, hns *hnswrapper.Hnsv2wrapperFake, expectedSetPolicies []*hcn.SetPolicySetting) bool {
+	cachedSetPolicies := hns.Cache.AllSetPolicies(azureNetworkID)
+
+	success := assert.Equal(t, len(expectedSetPolicies), len(cachedSetPolicies), "unexpected number of SetPolicies")
+	for _, expectedSetPolicy := range expectedSetPolicies {
+		cachedSetPolicy, ok := cachedSetPolicies[expectedSetPolicy.Id]
+		success = assert.True(t, ok, fmt.Sprintf("expected SetPolicy not found. ID %s, name: %s", expectedSetPolicy.Id, expectedSetPolicy.Name)) && success
+		if !ok {
+			continue
+		}
+
+		members := strings.Split(cachedSetPolicy.Values, ",")
+		sort.Strings(members)
+		copyOfCachedSetPolicy := *cachedSetPolicy
+		copyOfCachedSetPolicy.Values = strings.Join(members, ",")
+
+		success = assert.Equal(t, expectedSetPolicy, &copyOfCachedSetPolicy, fmt.Sprintf("SetPolicy has unexpected contents. ID %s, name: %s", expectedSetPolicy.Id, expectedSetPolicy.Name)) && success
+	}
+
+	return success
+}
+
+// verifyACLs is true if HNS strictly has the expected Endpoints and ACLs.
+func verifyACLs(t *testing.T, hns *hnswrapper.Hnsv2wrapperFake, expectedEndpointACLs map[string][]*hnswrapper.FakeEndpointPolicy) bool {
+	cachedEndpointACLs := hns.Cache.GetAllACLs()
+
+	success := assert.Equal(t, len(expectedEndpointACLs), len(cachedEndpointACLs), "unexpected number of Endpoints")
+	for epID, expectedACLs := range expectedEndpointACLs {
+		cachedACLs, ok := cachedEndpointACLs[epID]
+		success = assert.True(t, ok, fmt.Sprintf("expected ACL not found for endpoint %s", epID)) && success
+		if !ok {
+			continue
+		}
+
+		success = assert.Equal(t, len(expectedACLs), len(cachedACLs), fmt.Sprintf("unexpected number of ACLs for Endpoint with ID: %s", epID)) && success
+		for _, expectedACL := range expectedACLs {
+			foundACL := false
+			for _, cacheACL := range cachedACLs {
+				if expectedACL.ID == cacheACL.ID {
+					if reflect.DeepEqual(expectedACL, cacheACL) {
```

Comment:
`cmp.Equal` from `google.com/go-cmp/cmp` is categorically better than `reflect.DeepEqual`. The talk that lays out the argument for it is here: https://www.youtube.com/watch?v=OuT8YYAOOVI. There's also a `cmp.Diff` that will make it easier to see _why_ it's not equal.